File: frontrdc/src/app/gestioncategorie/service/categorie.service.ts
import { Injectable } from '@angular/core';
import { environment } from 'src/environments/environment';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class CategorieService {
  api = environment.apii;
  constructor(private http:HttpClient) { }
  
  allCategories(){
    return this.http.get(this.api+'categorie/list')
  }

  addCategorie(data){
    return this.http.post(this.api+'categorie/create',data)
  }

  updateCategorie(id,data){
    return this.http.post(this.api+'categorie/update/'+id,data)
  }

  deleteCategorie(element){
    return this.http.post(this.api+'categorie/delete',element)
  }
}

File: backendrdc/src/main/java/sn/gainde2000/pi/core/service/CategorieService.java
package sn.gainde2000.pi.core.service;

import java.util.List;

import sn.gainde2000.pi.core.entity.Categorie;

public interface CategorieService {
	public Categorie addCategorie(Categorie categorie);
	public List<Categorie> listCategories();
	public void updateCategorie(Long id,Categorie categorie);
	public void supprimerCategorie(Long id);
}
